src/record/error/types/parser.ts

```diff
@@ -0,0 +1,3 @@
+export interface IParser {
```

Comment:
Please use `type` instead of `interface`.
We can choose based on personal preference, but we use `type` now.

https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#differences-between-type-aliases-and-interfaces

Reply:
I think the name of this type should contain `KintoneAllRecordsError` like `KintoneAllRecordsErrorParser`.

Reply:
@mshrtsr 
 
When designing `KintoneAllRecordsErrorParser`, I referred to [this blog](https://blog.logrocket.com/types-vs-interfaces-in-typescript). Because, their implementations are objects(classes), not functions. So, that's why I choose `interface` instead of `type`.

But, maybe there is no problem with either `type` or `interface`. So, It's okay to use `type`.
I will refactor it. Thank you

Reply:
@hung-cybo 
Thank you to let me know!
Yes, we can change to `interface` if we need to use it like extending a class.

Reply:
@hung-cybo 
Sorry, I misunderstood your implementation.
`IParser` focuses on parsing `KintoneAllRecordsError.error` (`KintoneRestAPIError`), so it should be `KintoneRestAPIErrorParser`.

Reply:
@mshrtsr 

I changed it to `KintoneAllRecordsErrorParser` because I think It will be more extendable and updatable than `KintoneRestAPIErrorParser` (`KintoneAllRecordsError` gives more data than `KintoneRestAPIError`). So, what do you think about it?

Reply:
Now, the implementation of IParser is generating error messages from only KintoneRestAPIError.
I think both are ok at this time. so we keep it!